From a048522f64dbbe90cf77aea0c3f47cccdad50742 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Fri, 21 Jul 2023 19:30:40 +0200
Subject: [PATCH] Revert "Fix -static builds"

This reverts commit cfd7225fbb9507b2e443a494459bdaab5236d29d.

Upstream: https://github.com/openssl/openssl/issues/14574#issuecomment-817209126

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
[Dario: make the patch to be applied with fuzz factor 0]
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
---
 Configure | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Configure b/Configure
index ac3206e..764ef89 100755
--- a/Configure
+++ b/Configure
@@ -1509,10 +1509,6 @@ if ($config{prefix} && !$config{CROSS_COMPILE}) {
         unless file_name_is_absolute($config{prefix});
 }
 
-if (grep { $_ =~ /(?:^|\s)-static(?:\s|$)/ } @{$config{LDFLAGS}}) {
-    disable('static', 'pic', 'threads');
-}
-
 # Allow overriding the build file name
 $config{build_file} = env('BUILDFILE') || $target{build_file} || "Makefile";
 
@@ -1807,6 +1803,10 @@ if ($strict_warnings)
                 }
         }
 
+if (grep { $_ =~ /(?:^|\s)-static(?:\s|$)/ } @{$config{LDFLAGS}}) {
+    disable('static', 'pic', 'threads');
+}
+
 $config{CFLAGS} = [ map { $_ eq '--ossl-strict-warnings'
                               ? @strict_warnings_collection
                               : ( $_ ) }
-- 
2.43.0

